crates/binjs_io/src/json/write.rs

```diff
@@ -0,0 +1,352 @@
+//! A JSON writer for BinAST file structure.
+//! The written file is supposed to be filtered with some command to generate
```

Comment:
So what kind of filter are you expecting, exactly?

Reply:
The filter modifies a valid tree into a invalid tree, which can be encoded as a file, but it's expected to result in an error.
Something like, add unexpected field to some interface, remove field, put a subtree which is not allowed by syntax into somewhere.

The file is used to fuzz the decoder implementation, more efficiently than just mutating the binary file,
and also to keep testing specific scenario across file-format version bump.

I'll add comments.